pkg/cmd/check_banned.go

```diff
@@ -0,0 +1,115 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/banneduser"
+	"github.com/codeready-toolchain/toolchain-common/pkg/hash"
+	"github.com/kubesaw/ksctl/pkg/client"
+	clicontext "github.com/kubesaw/ksctl/pkg/context"
+	"github.com/kubesaw/ksctl/pkg/ioutils"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/errors"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func NewCheckBannedCommand() *cobra.Command {
+	var mur string
+	var signup string
+	var email string
+
+	command := &cobra.Command{
+		Use:   "check-banned [--mur|-m <mur-name>] | [--signup|-s <signup-name>] | [--email|-s <email>]",
+		Short: "Unban the user so that they can use the platform again",
+		Long:  "Unban the user that previously registered with the provided email so that they can start using the platform again.",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			term := ioutils.NewTerminal(cmd.InOrStdin, cmd.OutOrStdout)
+			ctx := clicontext.NewCommandContext(term, client.DefaultNewClient)
+			return CheckBanned(ctx, mur, signup, email)
+		},
+	}
+	command.Flags().StringVarP(&mur, "mur", "m", "", "the name of the master user record to check the banned status of")
+	command.Flags().StringVarP(&signup, "signup", "s", "", "the name of the signup to check the banned status of")
+	command.Flags().StringVarP(&email, "email", "e", "", "the email of the user to check the banned status of")
+
+	return command
+}
+
+func CheckBanned(ctx *clicontext.CommandContext, mur string, signup string, email string) error {
+	if !validateCheckBannedInput(mur, signup, email) {
+		return fmt.Errorf("exactly 1 of --mur, --signup or --email must be specified")
+	}
+
+	cl, cfg, err := getClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	if mur != "" {
+		email, err = findEmailByMUR(ctx, cl, mur, cfg.OperatorNamespace)
+	} else if signup != "" {
+		email, err = findEmailByUserSignup(ctx, cl, signup, cfg.OperatorNamespace)
+	}
+	if err != nil {
+		return err
+	}
+
+	if email == "" {
+		ctx.Println("User not found.")
+		return nil
+	}
+
+	emailHash := hash.EncodeString(email)
```

Comment:
As mentioned in the other comment, you need to get UserSignup anyway, because when the user is banned, then all other CRs are deleted. In this case, you can use `toolchain.dev.openshift.com/email-hash` label that is set in all UserSignups and already contains the hashed email.

Reply:
Thanks for the comments - I didn't know about the relationship between MUR name and UserSignup.Status.CompliantUsername. I kept the email hash computation inline though, because we'd have to do it in the case we've been given just the email. PTAL at the new version of the code, where I hope the comments explain why I didn't want to load the UserSignup unconditionally.